Update EIP-6963: Add RDNS section for motivation and requirements (#7464)
ethereum/EIPs: EIPS/eip-6963.md

The `icon` string MUST be a data URI as defined in [RFC-2397]. The image SHOULD be squared with 96x96px minimum resolution. The image format is RECOMMENDED to be either lossless or vector based such as PNG, WebP or SVG to make the image easy to render on the DApp. Since SVG images can execute Javascript, applications and libraries MUST render SVG images using the `<img>` tag to ensure no untrusted Javascript execution can occur.

#### RDNS

> The **`rdns`** (Reverse-DNS) property serves to provide an identifier which DApps can rely on to be stable between sessions. The Reverse Domain Name Notation is chosen to prevent namespace collisions.

Reviewer: ```suggestion
The **`rdns`** (Reverse-DNS) property serves to provide an identifier which DApps can rely on to be stable between sessions. The Reverse Domain Name Notation is chosen to prevent namespace collisions and to organize the namespace of these identifiers intuitively for developers.
```

The Reverse-DNS convention implies that the value should start with a reversed DNS domain name controlled by the Provider. The domain name should be followed by a subdomain or a product name. Example: `com.example.MyBrowserWallet`.

> - The DNS part of the `rdns` value MUST BE a valid [RFC-1034] Domain Name;

Reviewer: ```suggestion
- The `rdns` value MUST BE a valid [RFC-1034] Domain Name;
```
I feel like exempting the "subdomain or product name" segment from RFC-1034 just invites problems, while requiring product names to be encoded as valid subdomains will be a stitch in time if future verification specs define garden-variety `/well-known/` lookups :D

Author: Here I used the wording "DNS part of the `rdns`" only to mean the original non-reversed string.
We can't say that the literal `rnds` value needs to be a valid domain, because "com.example" isn't. It's the reversed-to-the original string that needs to be a valid domain :) 
I'm not sure how to say this without making the sentence unnecessarily long 

I agree that there's no need to exempt the "product name" part from being a valid part of the domain rfc.

At the same time I feel we have to highlight that, given a value of `com.example.wallet-name`, the following is expected:
* `example.com` SHOULD be an _active_ domain
* `wallet-name.example.com` MUST BE a _valid domain name_,
* `wallet-name.example.com` is NOT required to be an _active_ domain.

Perhaps this is better done with an example?

> - The DNS part of the `rdns` value SHOULD BE an active domain controlled by the Provider;

Reviewer: Likewise
```suggestion
- The root domain of the `rdns` value SHOULD BE an active domain controlled by the Provider;
```
Can't be a must because we don't provide a mechanism for making this testable, but hints that a future extension spec might.

Author: I don't think we've defined a "_root domain_" thought 🤔

Reviewer: ```suggestion
- The DNS part of the `rdns` value SHOULD BE an active apex domain controlled by the Provider;
```
Googling around, I realized "root" isn't really a stable or defined term.  This recent RFC supersedes the older ones and defines `apex` as the "highest delegation point owned by someone", basically: 
https://www.rfc-editor.org/rfc/rfc8499

- DApps MAY reject the Providers which do not follow the Reverse-DNS convention correctly;

> - DApps MUST NOT use the `rnds` value for feature detection;

Author: Should the spec also prescribe against prioritization?
For example, a DApp with a Connect Screen that chooses to show some wallet options at the top, and chooses to hide other wallets behind the "show more" button.
Such UI patterns _may_ lead to the "browser wars" problem, which this particular proposal is trying to avoid in the first place.

Reviewer: I believe that falls under "economic behaviors" and definitely hard to banish with a MUST NOT.  I would say it's better to phrase any guidance in technical terms and, at most, remind implementers that such design decisions create incentives for bad security, i.e. impersonation.  For example:

```suggestion
- DApps SHOULD NOT use the `rnds` value for feature detection as these are self-attested and prone to impersonation or bad incentives without an additional verification mechanism; feature-discovery and verification are both out of scope of this interface specification.
```

Author: Good point!
